SAC updates (#522)
hellohaptik/docs_haptik: docs/agent-chat/agent-reports.md

The Smart Agent Chat report gives you a list of all the agent conversations in a selected time period. You can now analyse your agents based on various metrics like response times and more on a granular level. Please go through each of the metrics mentioned below. We have described each column metric added in the report so that you could find out the best performing agents and monitor agent performance on a more granular level.

There are 3 reports which you will receive. 

> 1. Agent chat metrics

Reviewer: @g1-reddy Are the names correct here? Have we changed it?

Reviewer 2: @g1-reddy Minor point but we're calling the first csv "Agent Data", might need to address this

> 1. Agent chat metrics

Reviewer 3: We can also mention the name of the csv files? 
The format is as below 

1. Agent_data_Teamname_2021-05-26_12:54-2021-05-27_12:54.csv 
2. online_offline_Teamname_2021-05-26_12:54-2021-05-27_12:54.csv
3. Login_logout_Teamname_2021-05-26_12:54-2021-05-27_12:54.csv

    The values is `Yes` if the conversation was reassigned otherwise value will be `No`
    

> 9. **Reassigned**

Reviewer: @g1-reddy can you help us explain this column?

11. **First User Message**

    It is the message sent by the user that is the first message from the end user.
    

> 12. **Agent Abandoned**

Reviewer: @g1-reddy can you help us explain this too?

31. **Abandoned By User**

    When a conversation engagement is abandoned by the end user that is no user message sent while the agent has this conversation, it's counted as abandoned by user and we show a `Yes` value in this column
    
> Read about Abandoned metric [here](https://docs.haptik.ai/agent-chat/analytics#chat-stats)

32. **Delayed**

    Delayed time is set in the Team settings. For all instances, wherein the delay message is sent i.e. the delay time has passed when no agent being assigned to end user, we show a `Yes` string value in this column
    
> Read about Delayed metric [here](https://docs.haptik.ai/agent-chat/analytics#chat-stats)

33. **Messages Sent By Agent**

    Total number of messages which were sent by a specific agent to the end user
    
34. **Messages Sent by user**

    Total number of messages received from the end user.

## Agent Online/Offline report

![image](https://user-images.githubusercontent.com/75118325/122749859-776ffa80-d2ab-11eb-9750-682f3f7bccef.png)

1. **Agent Name**

    It is an agent's unique name who has handled conversations on the system.

    Examples: `gogo`, `saurabhsingh`

    ```

>     Note: `gogo` is an internal name for our system (bot)

Reviewer 3: @nitinm25  I don't think we will show gogo as one of the agents right?

Reviewer 2: Agent metrics reports include 'gogo' chats, but not the two new reports

Reviewer: So should we keep it as it is?

    ```
2. **Online date (Date)**

>     The date on which agent was online.

Reviewer 3: We have to address the null state scenario(Where this field might not have any data)
If the report was downloaded on June 1st but the agent was nine from May 31st, given that he was online before the selected date. the message would show "Agent was online before  <start date>"

> 3. **Online at (HH:MM:SS)**

Reviewer 3: Null state

"No Online time available"

    
    The timestamp at which the agent's status was changed to **Online**.

> 4. **Offline date(Date)**

Reviewer 3: Null state
"No Offline date available"

    The date on which the agent status was changed to offline.
    

> 5. **Offline at (HH:MM:SS)**

Reviewer 3: Null state

"Agent online as of "<End Timestamp>""

    The timestamp at which the agent's status was changed to **Offline**.
    

> 6. **Online Hours (HH:MM:SS)**

Reviewer 3: @nitinm25  can you give a small write up on the granularity of this data point?

Reviewer 2: The time an agent spent online, with the Agent Chat window active
Note: This metric has a a degree of accuracy of 150 seconds (2.5 minutes)

Reviewer: What do we mean by "Note: This metric has a a degree of accuracy of 150 seconds (2.5 minutes)" Should we add this in the next line?

Reviewer: already mentioned below, not sure why we should add accuracy degree.

    
    The total number of hours the agent was online.
    

> 7. **Marked online by**

Reviewer 3: it can be the agent himself or the TL and the name will be sown in this column

    
    This column specifies who changed the status of the agent to **Online**.
    
8. **Marked offline by**

>     This column specifies who changed the status of the agent to **Offline**.

Reviewer 3: This action can be taken by the agent himself or the TL or the system (after 2.5 min of inactivity by closing the chat window or by losing internet)

9. **Time zone**
    
    The timezone you have downloaded the report in.
    
## Agent Login/Logout report

> 

Reviewer 3: the same null states apply for these data points as well.

30. **User Comment**

    When the end user shares their conversation feedback by giving a rating, user can also add a text comment which will be recorded and visible in this column

31. **Abandoned By User**

    When a conversation engagement is abandoned by the end user that is no user message sent while the agent has this conversation, it's counted as abandoned by user and we show a `Yes` value in this column
    
> Read about Abandoned metric [here](https://docs.haptik.ai/agent-chat/analytics#chat-stats)

32. **Delayed**

    Delayed time is set in the Team settings. For all instances, wherein the delay message is sent i.e. the delay time has passed when no agent being assigned to end user, we show a `Yes` string value in this column
    
> Read about Delayed metric [here](https://docs.haptik.ai/agent-chat/analytics#chat-stats)

33. **Messages Sent By Agent**

    Total number of messages which were sent by a specific agent to the end user
    
34. **Messages Sent by user**

    Total number of messages received from the end user.

## Agent Online/Offline report

![image](https://user-images.githubusercontent.com/75118325/122749859-776ffa80-d2ab-11eb-9750-682f3f7bccef.png)

> 1. **Agent Name **

Reviewer: ```suggestion
1. **Agent Name**
```

>     It is an expert's unique name who has claimed this chat conversation.

Reviewer: ```suggestion
    It is an agent's unique name who has handled conversations on the system.
```

    Examples: `gogo`, `saurabhsingh`

    ```
    Note: `gogo` is an internal name for our system (bot)
    ```
2. **Online date (Date)**

    The date on which agent was online.

3. **Online at (HH:MM:SS)**
    

>     The timestamp at which the agent changes their status to **Online**.

Reviewer: ```suggestion
    The timestamp at which the agent status was changed to **Online**.
```

4. **Offline date(Date)**

>     The date at which the agent status was changed to offline.

Reviewer: ```suggestion
    The date on which the agent status was changed to offline.
```

    
5. **Offline at (HH:MM:SS)**

>     The timestamp at which the agents status was changed to **Offline**.

Reviewer: ```suggestion
    The timestamp at which the agent's status was changed to **Offline**.
```

    
6. **Online Hours (HH:MM:SS)**
    
    The total number of hours the agent was online.
    
7. **Marked online by**
    
    This column specifies who changed the status of the agent to **Online**.
    
8. **Marked offline by**

    This column specifies who changed the status of the agent to **Offline**.

9. **Time zone**
    
    The timezone you have downloaded the report in.
    
## Agent Login/Logout report

![image](https://user-images.githubusercontent.com/75118325/122755524-bf465000-d2b2-11eb-90c5-5855e7216034.png)

1. **Agent Name**
    

>     It is an expert's unique name who has claimed this chat conversation.

Reviewer: ```suggestion
    It is an agent’s unique name who has handled conversations on the system.
```
